src/layout/index.tsx

```diff
@@ -269,3 +248,92 @@ const Layout: React.FC<LayoutProps> = ({
 };
 
 export default Layout;
+
+function useTogglableBurgerMenu<T extends HTMLElement, U extends HTMLElement>(
+  modalRef: React.RefObject<T>,
+  ulRef: React.RefObject<U>,
+) {
+  const [isMenuVisible, setMenuVisible] = useState(false);
+  const [tabIndex, setTabIndex] = useState(0);
+
+  useEffect(() => {
+    setTabIndex(isMenuVisible ? 0 : -1);
+
+    // Avoid scrolling when menu is visible.
+    if (isMenuVisible) {
+      document.body.style.overflow = 'hidden';
+    } else {
+      document.body.style.overflow = 'initial';
+    }
+  }, [isMenuVisible]);
+
+  useEffect(() => {
+    const handleClickOutside = (e: MouseEvent) => {
+      if (!e.target || !ulRef.current?.contains(e.target as Node)) {
+        setMenuVisible(false);
+      }
+    };
+    document.body.addEventListener('click', handleClickOutside);
+    return () => document.body.removeEventListener('click', handleClickOutside);
+  }, [modalRef, ulRef]);
+
+  const handleTabKey = useCallback(
+    (e: KeyboardEvent) => {
+      const focusableModalElements =
+        modalRef.current?.querySelectorAll<HTMLElement>(
+          '[role="button"],a[href], button, textarea, input[type="text"], input[type="radio"], input[type="checkbox"], select',
+        ) ?? [];
+      const allFocusables =
+        document.querySelectorAll<HTMLElement>(
+          '[role="button"],a[href], button, textarea, input[type="text"], input[type="radio"], input[type="checkbox"], select',
+        ) ?? [];
+
+      const first = focusableModalElements[0];
+      const last = focusableModalElements[focusableModalElements.length - 1];
+      const next =
+        Array.from(allFocusables).find(
+          (_, i) => allFocusables[i - 1] === document.activeElement,
+        ) ?? null;
+      const previous =
+        Array.from(allFocusables).find(
+          (_, i) => allFocusables[i + 1] === document.activeElement,
+        ) ?? null;
+
+      // On normal tabbing. If next element is outside modal, jump to first element
+      if (!e.shiftKey && !modalRef.current?.contains(next)) {
+        first?.focus();
+        return e.preventDefault();
+      }
+
+      // On "reversed" tabbing. If previous element is outside modal, jump to last element
+      if (e.shiftKey && !modalRef.current?.contains(previous)) {
+        last?.focus();
+        return e.preventDefault();
+      }
+
+      // Normal tab flow.
```

Comment:
Misplaced comment or just me not understanding?

Reply:
Can specify better! Point is, no case hits, so just follow normal browser behaviour.

Reply:
Right, gotcha.